src/datepicker/styled-components.js

```diff
@@ -110,9 +131,52 @@ export const StyledDay = styled('div', (props: SharedStylePropsT) => {
         : 'inherit',
     backgroundColor: $selected
       ? colors.primary
-      : $isHovered || $isHighlighted
+      : $isHovered || $isHighlighted || $pseudoHighlighted || $pseudoSelected
         ? colors.primary100
         : 'transparent',
-    borderRadius: borders.useRoundedCorners ? borders.radius200 : '0px',
+    ...(borders.useRoundedCorners
```

Comment:
could this be broken out into its own function and unit tested? seems to work fine in practice though

Reply:
what i'm concerned about is causing regressions on tangentially related code changes